src/subscription_manager/gui/registergui.py

```diff
@@ -996,6 +1004,58 @@ def pre(self):
         return True
 
 
+# After registering, we can upload package profiles.
+# But... we don't have to do it immed after registering, it could
+# be in a later screen or a back ground task (user flow isn't
+# altered by results of package profile, and it's kind of slow, it
+# could happen in it's own thread while we go through other screens
+class PerformPackageProfileSyncScreen(NoGuiScreen):
+    screen_enum = UPLOAD_PACKAGE_PROFILE_PAGE
+
+    def __init__(self, reg_info, async_backend, facts, parent_window):
+        super(PerformPackageProfileSyncScreen, self).__init__(reg_info, async_backend, facts, parent_window)
+        self.pre_message = _("Uploading package profile")
+
+    def _on_update_package_profile_finished_cb(self, result, error=None):
+        if error is not None:
+            self.emit('register-error',
+                      REGISTER_ERROR,
+                      error)
+            # Allow failure on package profile uploads
+            self.emit('move-to-screen', SELECT_SLA_PAGE)
+            self.pre_done()
+            return
+
+        try:
+            if self.info.get_property('activation-keys'):
+                self.emit('move-to-screen', REFRESH_SUBSCRIPTIONS_PAGE)
+                self.pre_done()
+                return
+            # could/should we rely on RegisterWidgets register-finished to handle this?
+            # ie, detecting when we are 'done' registering
+            elif self.info.get_property('skip-auto-bind'):
+                self.pre_done()
+                return
+            # Or more likely, the server doesn't support package profile updates
+            # so we got a result of 0 and no error
+            else:
+                self.emit('move-to-screen', SELECT_SLA_PAGE)
+                self.pre_done()
+                return
```

Comment:
We can drop all these individual "self.pre_done()" and "return" statements, since every possible branch does it, and it's performed in the exception handler AND immediately following it.